### PR TITLE
Support for call history management messages

### PR DESCRIPTION
**Pull request by alber2510**

### CODE REVIEW

**Comment by alber2510** [comment]
trigger compilation

**Comment by alber2510** [comment]
trigger compilation

File: examples/qtmegachatapi/chatItemWidget.cpp [hunk 1]
```diff
@@ -161,9 +161,23 @@ void ChatItemWidget::updateToolTip(const megachat::MegaChatListItem *item, const
...
             break;
 
+        case megachat::MegaChatMessage::TYPE_CALL_ENDED:
+        {
+            QString qstring(item->getLastMessage());
+            QStringList stringList = qstring.split(0x01);
+            assert(stringList.size() >= 2);
+            lastMessage.append("Call started by: ")
+                       .append(senderHandle)
+                       .append(" Durration: ")
```
Comment (jgandres) [inline examples/qtmegachatapi/chatItemWidget.cpp]: Change for Duration

File: examples/qtmegachatapi/chatMessage.cpp [hunk 2]
```diff
@@ -244,6 +244,25 @@ std::string ChatMessage::managementInfoToString() const
...
+        for (unsigned int i = 0; i < handleList->size(); i++)
+        {
+            char *participant_64 = this->mChatWindow->mMegaApi->userHandleToBase64(handleList->get(i));
+            ret.append(participant_64).append(" ");
+            delete participant_64;
+        }
+
+        ret.append("\nDuration: ")
+           .append(std::to_string(mMessage->getDuration()))
+           .append( "secs TermCode: ")
```
Comment (jgandres) [inline examples/qtmegachatapi/chatMessage.cpp]: Remove space

File: src/megachatapi.h [hunk 3]
```diff
@@ -640,6 +641,15 @@ class MegaChatMessage
         REASON_NO_CHANGES           = 6     /// Edited message has the same content than original message
     };
 
+    enum
+    {
+        END_CALL_REASON_ENDED       = 1,    /// Call finished normally
+        END_CALL_REASON_REJECTED    = 2,    /// Call was rejected by callee
```
Comment (jgandres) [inline src/megachatapi.h]: Change callee for caller

File: examples/qtmegachatapi/chatItemWidget.cpp (hunk 1 above)
Comment (alber2510) [reply examples/qtmegachatapi/chatItemWidget.cpp]: Solved it

File: examples/qtmegachatapi/chatMessage.cpp (hunk 2 above)
Comment (alber2510) [reply examples/qtmegachatapi/chatMessage.cpp]: Solved it

File: src/megachatapi.h [hunk 4]
```diff
@@ -3355,6 +3408,17 @@ class MegaChatListItem
      *  - MegaChatMessage::TYPE_TRUNCATE: empty string
      *  - MegaChatMessage::TYPE_ALTER_PARTICIPANTS: empty string
      *  - MegaChatMessage::TYPE_PRIV_CHANGE: empty string
+     *  - MegaChatMessage::TYPE_CALL_ENDED: string set separated by ASCII character '0x01'
+     *      Structure: duration(seconds)'0x01'termCode'0x01'participants1'0x01'participants2'0x01'...
```
[the remark this answers is not in the document]
Comment (alber2510) [reply src/megachatapi.h]: It isn't necessary notify the callid at MegaChatListItem

**Pull request merged**